core/common/src/main/java/alluxio/PropertyKey.java

```diff
@@ -652,13 +678,27 @@ public static PropertyKey fromString(String keyStr) {
    * Constructs a configuration property.
    *
    * @param property String of this property
+   * @param defaultValue Default value of this property in compile time if not null
    */
-  PropertyKey(String property) {
-    mString = property;
+  PropertyKey(String property, Object defaultValue) {
+    mName = property;
+    if (defaultValue == null) {
+      mDefaultValue = null;
```

Comment:
I see your question. Still in `Configuration.get` if the returned the `null` value, `RuntimeException` is thrown. We didn't change that semantics